numpy/lib/function_base.py

```diff
@@ -4368,6 +4368,19 @@ def delete(arr, obj, axis=None):
             return new
 
     if isinstance(obj, (int, integer)) and not isinstance(obj, bool):
+        single_value = True
+    else:
+        single_value = False
+        _obj = obj
+        obj = np.asarray(obj)
+        if obj.size == 0 and not isinstance(_obj, np.ndarray):
+            obj = obj.astype(intp)
+
+        elif obj.shape == (1,):
+            obj = obj.item()
```

Comment:
`item()` makes me worry here...

Reply:
Does this patch change the behavior of ` np.delete(np.arange(10), np.array([1], dtype=object))`?

Reply:
I think `obj = obj.reshape(())` would avoid that shortcoming - please add a test!

Reply:
Could your clarify the behavior you are concerned about for  `np.delete(np.arange(10), np.array([1], dtype=object))`. I will change it to `obj = obj.reshape(())` either way but the behavior appears the same to me. What would you like to see in the test?

Also, is there anything else you would like to see tested?